### ISSUE TITLE
How to reproduce your result on M3ED supervised semantic segmentation?

### ISSUE THREAD

**Issue by Yutaro-Aze** [grasp-lyrl/fast-feature-fields#7]
Hello, 

I am writing to ask for guidance on reproducing the supervised semantic segmentation results on M3ED, specifically for the F3 backbone shown in Figure 5 (b) (mIoU: 66.28%).

First, I evaluated the provided EventFFSegFormer weights, which can be downloaded using the following download command, and `src/f3/tasks/segmentation/test.py` for inference:
```
bash scripts/download/download_models.sh pretrained_models/seg seg
```
However, my results were significantly higher than those reported in the paper:
|  | mIoU (%) | Accuracy (%) |
|--------|--------|--------|
| paper | 66.28 | 92.99 |
| mine | 79.27 | 96.35 |

Due to this discrepancy, I suspect the pretrained model might have data leakage. To address this, I attempted to fine-tune the model myself using the following steps:

1. **LabeI Generation**: I generated semantic labels for `car_urban_day_ucity_big_loop` using`src/f3/tasks/segmentation/InternImage/segmentation/m3ed_internimage.py` with the `src/f3/tasks/segmentation/InternImage/segmentation/configs/mapillary/upernet_internimage_xl_512x1024_80k_mapillary.py` config.
2. **Fine-tuning**: I trained the model using `pretrained_models/f3/patchff_fullcardaym3ed_small_20ms.pth` as the F3 weight with this command:
```
python src/f3/tasks/segmentation/train.py \
    --conf confs/segmentation/segformer_b3_trainm3ed_1280x720x20.yml \
    --compile \
    --amp \
    --wandb \
```
3. **Inference**: I ran the evaluation using `src/f3/tasks/segmentation/test.py` with `confs/segmentation/eval_segformer_b3_20ms_m3edtest.yml` config.
However, the model overfitted quickly, and the mIoU never reached the 60% mark.

<img width="699" height="311" alt="Image" src="https://github.com/user-attachments/assets/6abcf7a0-2f24-4a79-8149-5e00fe1e2221" />

Could you please provide details on the exact training setup or data splits used to achieve the 66.28% mIoU reported in the paper? Any advice on how to correctly reproduce these results would be greatly appreciated.

**Comment by Yutaro-Aze** [comment]
Hello @sudoRicheek, I appreciate if you have time to take a look into this question.
I basically would like to just reproduce the segmentation result that is reported in the paper. Maybe I am getting into too detailed, in which case I also would like to know.
Thank you so much for your guidance in advance.

**Comment by sudoRicheek** [comment]
@Yutaro-Aze You are correct about the 66 vs 79 mIoU discrepancy. The model I released is trained on the whole dataset. However, fine-tuning the model/training it from scratch should yield results close to what is presented in the paper.

1. Are you using the full m3ed training set, what is mentioned in the appendix of the paper: 

<img width="1429" height="1946" alt="Image" src="https://github.com/user-attachments/assets/b5fab690-6ca3-4879-afdf-b958a764ca23" />

The config you are using seems to be the correct one.

2. The F3 weights you are using are correct. I don't recommend using `compile` and `amp`, for training downstream tasks. Maybe you can try without those.

The data splits are the ones mentioned in the appendix or the train/test YAML files (they should be the same). Let me check if I can find and share the trained models for reproducibility.

Thanks for using the code!

**Comment by Yutaro-Aze** [comment]
@sudoRicheek Thanks for the reply! I am using the full m3ed dataset as mentioned.
I actually tried w/, w/o `compile`, but I kept `amp` enabled, so it might be the cause. I'll give it a try without it.
Thanks again for your help.

**Comment by sudoRicheek** [comment]
@Yutaro-Aze Thanks for trying! Do let me know if it works. I will try to check if there are issues preventing reproducibility.

**Comment by Yutaro-Aze** [comment]
@sudoRicheek Hello, again. Following your suggestion, I disabled both `compile` and `amp`. However, unfortunately, I was still unable to reproduce the reported results.

|  | mIoU(%) | Accuracy(%) |
|--------|--------|--------|
| paper | 66.28 | 92.99 |
| mine(w/ amp, compile) | 48.86 | 86.56 |
| mine(w/o amp, compile) | 51.32 | 87.15 | 

Could you check the problem? If possible, I would appreciate it if you could share the detailed steps for creating the labels and the inference commands used. Also, please let me know if you have any questions regarding my implementation.

**Comment by sudoRicheek** [comment]
Sorry @Yutaro-Aze. I will get to it and check what the issue is. You seem to using it correctly. Here's the wandb run where I get 66.28/92.99:

<img width="2164" height="942" alt="Image" src="https://github.com/user-attachments/assets/3f99e7a5-45f1-4cf6-a7e7-72d262d3e4ef" />

**Comment by Yutaro-Aze** [comment]
Hi @sudoRicheek, following up on this issue - I've now attempted to reproduce the DSEC segmentation results from Figure 5(a) ("Trained and tested on DSEC"), and I'm seeing the same overfitting pattern.

### Setup
**F3 backbone**: downloaded patchff_fulldsec_small_20ms.pth via scripts/download/download_models.sh and placed it as follows:
```
bash scripts/download/download_models.sh outputs/pretrained_ff f3
mkdir -p outputs/patchff_fulldsec_small_20ms/models
ln -s ../../pretrained_ff/patchff_fulldsec_small_20ms.pth \
   outputs/patchff_fulldsec_small_20ms/models/last.pth
ln -s ../../../confs/ff/modeloptions/640x480x20_patchff_ds1_small.yml \
   outputs/patchff_fulldsec_small_20ms/models/config.yml
```
**Training command**:
```
python src/f3/tasks/segmentation/train.py \
    --conf confs/segmentation/segformer_b3_traindsec_640x480x20.yml \
    --name seg_f3_dsec_20ms
```
**Evaluation command**:
```
python src/f3/tasks/segmentation/test.py \
    --name seg_f3_dsec_20ms \
    --model best_miou \
    --useconfig confs/segmentation/eval_segformer_b3_20ms_fulldsecval_dsec.yml
```
### Results
| | mIoU(%) | Accuracy(%) |
|--------|--------|--------|
| paper | 55.41 | 91.91 |
| mine | 49.64 | 89.99 |

The model overfits severely — train mIoU reaches 91% while val mIoU stays flat at ~49% from epoch 9 through 99:
| Epoch | Train loss | Train mIoU(%) | Val loss | Val mIoU(%) |
|--------|--------|--------|--------|--------|
| 9 | 0.089 | 78.5 | 0.404 | 49.3 |
| 29 | 0.050 | 86.2 | 0.571 | 49.6 |
| 49 | 0.040 | 88.7 | 0.695 | 49.0 |
| 99 | 0.030 | 91.0 | 0.885 | 49.0 | 

<img width="1395" height="805" alt="Image" src="https://github.com/user-attachments/assets/5ae5b7bb-ca22-4477-b4b0-43cb7663bd77" />

Val loss monotonically increases while val mIoU never improves beyond the initial level.

Is there anything wrong with my setup, and what should I change to reproduce the reported results?                                                                                                   

Thanks!

**Comment by sudoRicheek** [comment]
Hmm, interesting, val miou should increase when using F3, since they lead to better generalization. This is quite weird. I am providing the trained M3ED segmentation model to reproduce the results in Fig 5.b.

Models are here: [**Google Drive Link**](https://drive.google.com/drive/folders/1_4HJA7l0I_xQGlWg5yVsEoZsFKaBZaJY?usp=drive_link)

You can clone the repo and place the weights in the `outputs` directory inside the repo, like this:

`gdown --folder 'https://drive.google.com/drive/folders/1_4HJA7l0I_xQGlWg5yVsEoZsFKaBZaJY?usp=sharing' -O outputs`

Then you can do the following to reproduce M3ED results 5.b,
`python3 src/f3/tasks/segmentation/test.py --name segformer_b3_trainm3ed_1280x720x20 --useconfig confs/segmentation/eval_segformer_b3_20ms_m3edtest.yml`
